TheThingsNetwork/lorawan-stack
# Accept and convert MAC address to Gateway EUI

#### Summary

Closes [#4149](https://github.com/TheThingsNetwork/lorawan-stack/issues/4149)

#### Changes

- Change the validation of the **Gateway EUI** input to accept 16 and 12 characters
- use `useDebounce` for the EUI and monitor the value having only 12 characters
- After 3 seconds with 12 characters, display warning message and a button on the input field
- Auto-fill **Gateway EUI** field by including `FF FE` value in the middle when clicking on the button

Initial:
![image](https://github.com/TheThingsNetwork/lorawan-stack/assets/49171937/65da761d-8d98-4322-9d41-d8809ac934c0)

After 3 seconds:
![image](https://github.com/TheThingsNetwork/lorawan-stack/assets/49171937/141cd647-e7f2-4e7a-a180-e65240d5a282)

After clicking on Convert MAC to EUI button:
![image](https://github.com/TheThingsNetwork/lorawan-stack/assets/49171937/db9cbc4f-5be7-435b-9cab-fc9f7c300786)

#### Testing

1. Log in in Console.
2. Go to Gateways in the navigation menu.
3. Click on Register gateway button.
4. Type 12 characters in the Gateway EUI field
5. After 3 seconds a warning message "It looks like you have entered a MAC address instead" and a button "Convert MAC to EUI" should show up
6. Clicking on "Convert MAC to EUI" button should auto-fill the Gateway EUI field with FF FE characters in the middle
7. The warning message and the button should disappear

#### Notes for Reviewers

Allowed 12 characters in the `validation-schema.js` as well because the error message has a bigger priority then the warning message and without this, `onBlur` on 12 characters the error message overrides the warning message.

#### Checklist
